pkg/printers/internalversion/describe.go

```diff
@@ -2875,6 +2875,24 @@ func describeHorizontalPodAutoscaler(hpa *autoscaling.HorizontalPodAutoscaler, e
 				w.Write(LEVEL_0, "failed to check Replication Controller\n")
 			}
 		}
+		if strings.ToLower(hpa.Spec.ScaleTargetRef.Kind) == "replicaSet" {
+			w.Write(LEVEL_0, "ReplicaSet pods:\t")
+			rs, err := d.client.Extensions().ReplicaSets(hpa.Namespace).Get(hpa.Spec.ScaleTargetRef.Name, metav1.GetOptions{})
+			if err == nil {
+				w.Write(LEVEL_0, "%d current / %d desired\n", rs.Status.Replicas, rs.Spec.Replicas)
+			} else {
+				w.Write(LEVEL_0, "failed to check ReplicaSet\n")
+			}
+		}
+		if strings.ToLower(hpa.Spec.ScaleTargetRef.Kind) == "deployment" {
```

Comment:
Same here. Don't add hardcoded scale targets. Replace the existing one with a call using the polymorphic scale client

Reply:
I find we can use subsource to show the current replicas and desired replicas, so I use it instead. =)